networking_oneview/ml2/drivers/oneview/common.py

```diff
@@ -540,3 +543,26 @@ def remove_inconsistence_from_db(
     database_manager.delete_oneview_network_lig(
         session, oneview_network_id=oneview_network_id
     )
+
+
+def delete_outdated_flat_mapped_networks(connection, flat_net_mappings):
+    session = get_session(connection)
+    mappings = flat_net_mappings.values()
+    mapped_networks_uuids = list(chain.from_iterable(mappings))
+    oneview_networks_uuids = (
+        network.oneview_network_id for network
+        in database_manager.list_neutron_oneview_network(session)
+        if not network.manageable)
+    unmapped_networks_uuids = (
+        uuid for uuid
+        in oneview_networks_uuids
+        if uuid not in mapped_networks_uuids)
+    for uuid in unmapped_networks_uuids:
+        database_manager.delete_neutron_oneview_network(
+            session, oneview_network_id=uuid)
+
+
+def get_session(connection):
```

Comment:
I think we can get the connection here: connection = CONF.database.connection. So we dont need to pass as parameter in every method. wdyt?

Reply:
I would suggest more, maybe changing the name of the method from get_session to get_database_session, I think it would be more clear. The only connection we use the database connection. wdyt?

Reply:
lgtm